### PR TITLE
[openimageio] Remove vcpkg_fail_port_install.

### PR DESCRIPTION
**Pull request by BillyONeal**
Separated out because it adds a supports expression to a feature. This port is not named in ci.baseline.txt.

In support of https://github.com/microsoft/vcpkg/pull/21502

### CODE REVIEW

**Comment by dg0yt** [comment]
Is there a hard hdf5 dependency? I guess it is wrong spelling in flann, #22731.

**Comment by BillyONeal** [comment]
> Is there a hard hdf5 dependency? I guess it is wrong spelling in flann, #22731.

I don't know. It's unlikely to be that issue though because flann is not in this dependency graph:

>PS C:\Dev\vcpkg2> .\vcpkg.exe depend-info openimageio   
boost-uninstall: 
boost-vcpkg-helpers: boost-uninstall
boost-config: boost-vcpkg-helpers
boost-assert: boost-config, boost-vcpkg-helpers
boost-throw-exception: boost-assert, boost-config, boost-vcpkg-helpers
boost-static-assert: boost-config, boost-vcpkg-helpers
boost-predef: boost-vcpkg-helpers
boost-preprocessor: boost-vcpkg-helpers
boost-type-traits: boost-config, boost-static-assert, boost-vcpkg-helpers
boost-io: boost-config, boost-vcpkg-helpers
boost-core: boost-assert, boost-config, boost-static-assert, boost-throw-exception, boost-vcpkg-helpers
boost-winapi: boost-config, boost-predef, boost-vcpkg-helpers
boost-utility: boost-config, boost-core, boost-io, boost-preprocessor, boost-throw-exception, boost-type-traits, boost-vcpkg-helpers
boost-integer: boost-assert, boost-config, boost-core, boost-static-assert, boost-throw-exception, boost-vcpkg-helpers
boost-detail: boost-config, boost-core, boost-preprocessor, boost-static-assert, boost-type-traits, boost-vcpkg-helpers
boost-mpl: boost-config, boost-core, boost-predef, boost-preprocessor, boost-static-assert, boost-type-traits, boost-utility, boost-vcpkg-helpers
boost-move: boost-assert, boost-config, boost-core, boost-static-assert, boost-vcpkg-helpers, boost-winapi
boost-function-types: boost-config, boost-core, boost-detail, boost-mpl, boost-preprocessor, boost-type-traits, boost-vcpkg-helpers
boost-typeof: boost-config, boost-preprocessor, boost-type-traits, boost-vcpkg-helpers
boost-tuple: boost-config, boost-core, boost-static-assert, boost-type-traits, boost-vcpkg-helpers
boost-container-hash: boost-assert, boost-config, boost-core, boost-detail, boost-integer, boost-static-assert, boost-type-traits, boost-vcpkg-helpers
boost-smart-ptr: boost-assert, boost-config, boost-core, boost-move, boost-static-assert, boost-throw-exception, boost-type-traits, boost-vcpkg-helpers
vcpkg-cmake:
boost-optional: boost-assert, boost-config, boost-core, boost-detail, boost-move, boost-predef, boost-static-assert, boost-throw-exception, boost-type-traits, boost-utility, boost-vcpkg-helpers
boost-fusion: boost-config, boost-container-hash, boost-core, boost-function-types, boost-mpl, boost-preprocessor, boost-static-assert, boost-tuple, boost-type-traits, boost-typeof, boost-utility, boost-vcpkg-helpers
boost-modular-build-helper: boost-uninstall, vcpkg-cmake
boost-conversion: boost-assert, boost-config, boost-core, boost-smart-ptr, boost-throw-exception, boost-type-traits, boost-typeof, boost-vcpkg-helpers
boost-concept-check: boost-config, boost-preprocessor, boost-static-assert, boost-type-traits, boost-vcpkg-helpers
boost-intrusive: boost-assert, boost-config, boost-container-hash, boost-core, boost-move, boost-static-assert, boost-vcpkg-helpers
boost-build: boost-uninstall
boost-array: boost-assert, boost-config, boost-core, boost-static-assert, boost-throw-exception, boost-vcpkg-helpers
boost-iterator: boost-assert, boost-concept-check, boost-config, boost-conversion, boost-core, boost-detail, boost-function-types, boost-fusion, boost-mpl, boost-optional, boost-smart-ptr, boost-static-assert, boost-type-traits, boost-utility, boost-vcpkg-helpers
boost-regex: boost-assert, boost-build, boost-concept-check, boost-config, boost-container-hash, boost-core, boost-integer, boost-modular-build-helper, boost-mpl, boost-predef, boost-smart-ptr, boost-static-assert, boost-throw-exception, boost-type-traits, boost-vcpkg-helpers, vcpkg-cmake
boost-mp11: boost-vcpkg-helpers
boost-container: boost-assert, boost-build, boost-config, boost-core, boost-intrusive, boost-modular-build-helper, boost-move, boost-static-assert, boost-type-traits, boost-vcpkg-helpers, boost-winapi, vcpkg-cmake
boost-bind: boost-config, boost-core, boost-vcpkg-helpers
boost-type-index: boost-config, boost-container-hash, boost-core, boost-preprocessor, boost-static-assert, boost-throw-exception, boost-type-traits, boost-vcpkg-helpers
boost-rational: boost-assert, boost-config, boost-core, boost-integer, boost-static-assert, boost-throw-exception, boost-type-traits, boost-utility, boost-vcpkg-helpers
boost-range: boost-array, boost-assert, boost-concept-check, boost-config, boost-container-hash, boost-conversion, boost-core, boost-detail, boost-iterator, boost-mpl, boost-optional, boost-preprocessor, boost-regex, boost-static-assert, boost-tuple, boost-type-traits, boost-utility, boost-vcpkg-helpers
boost-coroutine: boost-assert, boost-build, boost-config, boost-context, boost-core, boost-exception, boost-modular-build-helper, boost-move, boost-system, boost-throw-exception, boost-type-traits, boost-utility, boost-vcpkg-helpers, vcpkg-cmake       
boost-chrono: boost-assert, boost-build, boost-config, boost-core, boost-integer, boost-modular-build-helper, boost-move, boost-mpl, boost-predef, boost-ratio, boost-static-assert, boost-system, boost-throw-exception, boost-type-traits, boost-typeof, boost-utility, boost-vcpkg-helpers, boost-winapi, vcpkg-cmake
libde265: vcpkg-cmake, vcpkg-cmake-config
libjpeg-turbo: vcpkg-cmake, vcpkg-cmake-config
liblzma: vcpkg-cmake, vcpkg-cmake-config
zlib:szip: vcpkg-cmake, vcpkg-cmake-config
boost-atomic: boost-align, boost-assert, boost-build, boost-config, boost-modular-build-helper, boost-predef, boost-preprocessor, boost-static-assert, boost-type-traits, boost-vcpkg-helpers, boost-winapi, vcpkg-cmake
x265:
boost-asio: boost-align, boost-array, boost-assert, boost-bind, boost-chrono, boost-config, boost-core, boost-coroutine, boost-date-time, boost-function, boost-regex, boost-smart-ptr, boost-system, boost-throw-exception, boost-type-traits, boost-utility, boost-vcpkg-helpers
boost-filesystem: boost-assert, boost-atomic, boost-build, boost-config, boost-container-hash, boost-core, boost-detail, boost-io, boost-iterator, boost-modular-build-helper, boost-predef, boost-smart-ptr, boost-system, boost-type-traits, boost-vcpkg-helpers, boost-winapi, vcpkg-cmake
boost-stacktrace: boost-array, boost-build, boost-config, boost-container-hash, boost-core, boost-modular-build-helper, boost-predef, boost-static-assert, boost-type-traits, boost-vcpkg-helpers, boost-winapi, vcpkg-cmake
boost-foreach: boost-config, boost-core, boost-iterator, boost-mpl, boost-range, boost-type-traits, boost-vcpkg-helpers
boost-thread: boost-algorithm, boost-assert, boost-atomic, boost-bind, boost-build, boost-chrono, boost-concept-check, boost-config, boost-container, boost-container-hash, boost-core, boost-date-time, boost-exception, boost-function, boost-intrusive, boost-io, boost-iterator, boost-lexical-cast, boost-modular-build-helper, boost-move, boost-optional, boost-predef, boost-preprocessor, boost-smart-ptr, boost-static-assert, boost-system, boost-throw-exception, boost-tuple, boost-type-traits, boost-utility, boost-vcpkg-helpers, boost-winapi, vcpkg-cmake
fmt: vcpkg-cmake, vcpkg-cmake-confighdf5[szip, zlib]: szip, vcpkg-cmake, vcpkg-cmake-config, zlib
libheif: libde265, vcpkg-cmake, vcpkg-cmake-config, x265
libpng: vcpkg-cmake, vcpkg-cmake-config, zlib
openexr: zlib
robin-map: vcpkg-cmake, vcpkg-cmake-config
tiff[jpeg, lzma, zip]: libjpeg-turbo, liblzma, vcpkg-cmake, zlib
boost-random: boost-array, boost-assert, boost-build, boost-config, boost-core, boost-dynamic-bitset, boost-integer, boost-io, boost-modular-build-helper, boost-range, boost-static-assert, boost-system, boost-throw-exception, boost-type-traits, boost-utility, boost-vcpkg-helpers, vcpkg-cmake
libsquish: vcpkg-cmake, vcpkg-cmake-configopenimageio: boost-algorithm, boost-asio, boost-config, boost-filesystem, boost-foreach, boost-random, boost-regex, boost-smart-ptr, boost-stacktrace, boost-static-assert, boost-system, boost-thread, boost-type-traits, fmt, hdf5, libheif, libjpeg-turbo, libpng, libsquish, openexr, robin-map, tiff, vcpkg-cmake, vcpkg-cmake-config, zlib

**Comment by dg0yt** [comment]
> It's unlikely to be that issue though because flann is not in this dependency graph:

This doesn't say to much because `vcpkg ci` also builds ports which depend on the changed port, and pull in their entire universe of dependencies. And due to different feature configurations, this might uncover bugs in now-unchanged ports. 

The hdf5 issue also caught this fmt PR: #22707. And you probably wouldn't simply add hdf5 as a fmt dependency...

**Review by dg0yt — CHANGES_REQUESTED** [review]
No hdf5 dependency until proper research.

File: ports/openimageio/vcpkg.json [hunk 1]
```diff
@@ -49,6 +51,7 @@
     },
     "field3d": {
       "description": "Enable field3D support for openimageio",
+      "supports": "!windows & !uwp",
```
Comment (strega-nil-ms) [inline ports/openimageio/vcpkg.json:54]:
```suggestion
      "supports": "!windows",
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Waiting on @dg0yt's research.

File: ports/openimageio/vcpkg.json [hunk 2]
```diff
@@ -18,6 +19,7 @@
     "boost-thread",
     "boost-type-traits",
     "fmt",
+    "hdf5",
```
Comment (strega-nil-ms) [inline ports/openimageio/vcpkg.json:22]: This is at least wrong on osx, since I can install openimageio:x64-osx without hdf5

**Comment by BillyONeal** [comment]
> The hdf5 issue also caught this fmt PR: #22707. And you probably wouldn't simply add hdf5 as a fmt dependency...

Admittedly I know almost nothing about hdf5 :)

**Comment by dg0yt** [comment]
> No hdf5 dependency until proper research.

I didn't want to do that myself, because the ports I'm interested make enough work. 
Anyways: hdf5 is dependency of port field3, which should be hidden behind the field3 feature. 

> > Is there a hard hdf5 dependency? I guess it is wrong spelling in flann, #22731.
>
> It's unlikely to be that issue though because flann is not in this dependency graph

Inspecting the log of the build errror confirmed my guess:
~~~
-- Check for Flann
CMake Error at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:788 (_find_package):
  By not providing "Findhdf5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "hdf5", but
  CMake did not find one.

  Could not find a package configuration file provided by "hdf5" with any of
  the following names:

    hdf5Config.cmake
    hdf5-config.cmake

  Add the installation prefix of "hdf5" to CMAKE_PREFIX_PATH or set
  "hdf5_DIR" to a directory containing one of the above files.  If "hdf5"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /mnt/vcpkg-ci/installed/x64-linux/share/flann/flann-config.cmake:31 (find_package)
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:788 (_find_package)
  CMakeLists.txt:141 (find_package)
~~~

**Comment by BillyONeal** [comment]
> > No hdf5 dependency until proper research.
> 
> I didn't want to do that myself, because the ports I'm interested make enough work. Anyways: hdf5 is dependency of port field3, which should be hidden behind the field3 feature.

Yeah sorry, I was digging into that but had not gotten there yet. But Thanks!

I will wait for https://github.com/microsoft/vcpkg/pull/22731

**Pull request merged**